cmd/replay.go

```diff
@@ -0,0 +1,208 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/odpf/optimus/core/set"
+
+	pb "github.com/odpf/optimus/api/proto/odpf/optimus"
+	"github.com/odpf/optimus/config"
+	"github.com/olekukonko/tablewriter"
+	"github.com/pkg/errors"
+	cli "github.com/spf13/cobra"
+	"github.com/xlab/treeprint"
+	"google.golang.org/grpc"
+)
+
+type taskRunBlock struct {
+	name   string
+	height int
+	runs   set.Set
+}
+
+func taskRunBlockComperator(a, b interface{}) int {
+	aAsserted := a.(taskRunBlock)
+	bAsserted := b.(taskRunBlock)
+	switch {
+	case aAsserted.height < bAsserted.height:
+		return -1
+	case aAsserted.height > bAsserted.height:
+		return 1
+	}
+	return strings.Compare(aAsserted.name, bAsserted.name)
+}
+
+//formatRunsPerDAGInstance returns a hashmap with DAG -> Runs[] mapping
+func formatRunsPerDAGInstance(instance *pb.ReplayResponseNode, taskReruns map[string]taskRunBlock, height int) {
+	if _, ok := taskReruns[instance.JobName]; !ok {
+		taskReruns[instance.JobName] = taskRunBlock{
+			name:   instance.JobName,
+			height: height,
+			runs:   set.NewTreeSetWithTimeComparator(),
+		}
+	}
+
+	for _, taskRun := range instance.Runs {
+		parsedTime, _ := time.Parse(time.RFC3339, taskRun)
```

Comment:
if taskRun is just a time.RFC3339 timestamp, why not use timestamp protobuf for this? Why string?

Reply:
I have made this change